### ISSUE TITLE
caption on the bottom of a longtable

### ISSUE THREAD

**Issue by schlagi123** [latex3/latex2e#90]
!! Please fill out all sections after reading CONTRIBUTING guidelines (and remove lines starting with !!)

## Brief outline of the bug

Hello,

I must use a longtable and have the requirement that every table captions is behind the table. Currently this is not possible with a longtable. The problem is, that a \endfirsthead command is missing.

## Minimal example showing the bug

```latex
\RequirePackage{latexbug}
\documentclass{article}

\usepackage{longtable,blindtext}

\begin{document}
	\begin{longtable}{p{\textwidth}}
		\caption{Test} \label{t:test} \\
		\endfirsthead
		\caption*{\tablename{} \ref{t:test} (continue)} \\
		\endhead
		%\caption{Test} \label{t:test} \\
		%\endfirstfoot % does not exist
		\caption*{\tablename{} \ref{t:test} (continue)} \\
		\endfoot		
		\hline\blindtext \\ 
		\hline\blindtext \\ 
		\hline\blindtext \\
		\hline\blindtext \\
		\hline\blindtext \\
		\hline\blindtext \\
		\hline\blindtext \\
		\hline\blindtext \\
		\hline\blindtext \\														
		\hline\blindtext \\
		\hline\blindtext \\
		\hline\blindtext \\
		\hline
	\end{longtable}
\end{document}
```

**Comment by eg9** [comment]
Are your referring to the fact that `\endfirstfoot` does not exist, rather than `\endfirsthead` which _does_ exist?

**Comment by schlagi123** [comment]
Yes, `\endfirsthead`, `\endhead`, `\endfoot` and `\endlastfoot` are exist.

**Comment by davidcarlisle** [comment]
This was really by design but the first hit for a search for \endfirstfoot has some code from Heiko Oberdiek that probably works in your case


https://tex.stackexchange.com/questions/68439/caption-at-foot-of-long-tables-longtable-package

There are some issues in general, especially with "short" longtables as because of usual issues with the asynchronous nature of TeX's output routines longtable doesn't always reliably know if the first page is in fact also the last, however in this context perhaps that does not matter.

Heiko's code is already listed for possible inclusion in a future update

http://htmlpreview.github.io/?https://raw.githubusercontent.com/davidcarlisle/dpctex/master/longtable/ltbugs.html

David

**Comment by schlagi123** [comment]
You are ride, this is a solution.

> Heiko's code is already listed for possible inclusion in a future update

When and by whom this code will be included? I could include this and create a pull request if it is ok?

**Comment by staerz** [comment]
Hi there,

I'm having a similar problem.

In particular I'm using your longtable to set up a few given table templates that aught to define a fixed header.

The template I'm trying to define looks like that:

```tex
  \newenvironment{templatetable}[6]{
    \def\templatetableshortcaption{#1}
    \def\templatetablecaption{#2}
    \def\templatetablelabel{#3}

    \setlength\LTleft{0pt}
    \setlength\LTright{0pt}
    \begin{longtable}{@{\extracolsep{\fill}}#5@{}}
      \hline
      #6 \\
      \hline
      \endfirsthead
      \caption[]{...continued from previous page} \\
      \hline
      #6 \\
      \hline
      \endhead
      \hline % some more vertical space would be nice
      \caption*{...continued on next page} \\
      \endfoot
      \endlastfoot
  }{
      \hline
      \caption[\templatetableshortcaption]{\templatetablecaption}
      \label{tab:\templatetablelabel}
    \end{longtable}
  }
```

The non-obvious parameters are:

- 4: not used
- 5: column definitions (e.g. `|L{.47\linewidth}|L{.47\linewidth}|`)
- 6: column header (e.g. `\textbf{Name or Number} & \textbf{Description}`)

Now that all works pretty fine, except for an edge case where the table would just fit the bottom of the page - and for no apparent reason, the header is repeated on the next page (but no further body follows as the table is done), see the picture attached.

![longtable_pagebreak](https://user-images.githubusercontent.com/60404042/106664909-ab07ed80-6573-11eb-943c-cb229dc2fdff.png)

So I'm wonding if I'm doing a stupid mistake when setting up this template (order of commands) or if it's really a bug of `longtable`.
